docs/user/reference/general/toggling-upload-status.md

```diff
@@ -1,62 +1,134 @@
-
-{{#include ../../../shared/blockquote-warning.html}}
+```Kotlin
+import mozilla.telemetry.glean.Glean
 
-##### Do not call `setUploadEnabled` before initializing
+open class MainActivity : AppCompatActivity() {
+    override fun onCreate() {
+        // ...
 
-> If called before `Glean.initialize()` the call to `Glean.setUploadEnabled()` will be ignored.
-> Set the initial state using `uploadEnabled` on `Glean.initialize()`.
+        uploadSwitch.setOnCheckedChangeListener { _, isChecked ->
+            if (isChecked) {
+                Glean.setUploadEnabled(true)
+            } else {
+                Glean.setUploadEnabled(false)
+            }
+        }
+    }
+}
+```
 
 </div>
+<div data-lang="Java" class="tab">
+
+```Java
+import mozilla.telemetry.glean.Glean
 
+Glean.INSTANCE.setUploadEnabled(false);
+```
+
+</div>
 <div data-lang="Swift" class="tab">
 
-`Glean.shared.setUploadEnabled()` should be called in response to the user enabling or disabling telemetry.
 
-{{#include ../../../shared/blockquote-warning.html}}
+```Swift
+import Glean
+import UIKit
 
-##### Do not call `setUploadEnabled` before initializing
+class ViewController: UIViewController {
+    @IBOutlet var enableSwitch: UISwitch!
 
-> If called before `Glean.shared.initialize()` the call to `Glean.shared.setUploadEnabled()` will be ignored.
-> Set the initial state using `uploadEnabled` on `Glean.shared.initialize()`.
+    // ...
 
-</div>
+    @IBAction func enableToggled(_: Any) {
+        Glean.shared.setUploadEnabled(enableSwitch.isOn)
+    }
+}
+```
 
+
+</div>
 <div data-lang="Python" class="tab">
 
-`Glean.set_upload_enabled()` should be called in response to the user enabling or disabling telemetry.
+```python
+from glean import Glean
+
+Glean.set_upload_enabled(false)
+```
 
-{{#include ../../../shared/blockquote-warning.html}}
+</div>
+<div data-lang="Rust" class="tab">
 
-##### Do not call `set_upload_enabled` before initializing
+```Rust
+use glean;
 
-> If called before `Glean.initialize()` the call to `Glean.set_upload_enabled()` will be ignored.
-> Set the initial state using `upload_enabled` on `Glean.initialize()`.
+glean::set_upload_enabled(false);
+```
 
 </div>
-
 <div data-lang="JavaScript" class="tab">
 
-`Glean.setUploadEnabled()` should be called in response to the user enabling or disabling telemetry.
-
-{{#include ../../../shared/blockquote-warning.html}}
 
-##### Do not call `setUploadEnabled` before initializing
+```js
+import Glean from "@mozilla/glean/web";
 
-> If called before `Glean.initialize()` the call to `Glean.setUploadEnabled()` will be ignored.
-> Set the initial state using `uploadEnabled` on `Glean.initialize()`.
+const uploadSwitch = document.querySelector("input[type=checkbox].upload-switch");
+uploadSwitch.addEventListener("change", event => {
+    if (event.target.checked) {
+        Glean.setUploadEnabled(true);
+    } else {
+        Glean.setUploadEnabled(false);
+    }
```

Comment:
same here I guess (though I do understand that explicitly having both if branches might be easier to read.

```suggestion
    Glean.setUploadEnabled(event.target.checked);
```

Reply:
I think since this is just the docs I'd rather leave both branches.